openslides/core/static/js/core/start.js

```diff
@@ -46,6 +47,27 @@ angular.module('OpenSlidesApp.core.start', [])
                 this.bootup();
             },
         };
+
+        // We need to 'ping' the server with a get request to whoami, because then we can decide,
+        // if the server is down or respond with a 403 (this cannot be differentiated with websockets)
+        autoupdate.registerRetryConnectCallback(function () {
+            return $q(function (resolve) {
```

Comment:
I understand this approach but I think the current solution is not consistent: Rejecting a promise is the async way of throwing an error.

You can simplify your code if you consider the check here not as check of connection established and session flush but as check if OpenSlides has to be shut down (i. e. the retry process has to be stopped).

This pseudo code is only to explain how we can consider this case.
```javascript
return $http().then(function (success) {
    if !(......) {
        // Error because we expected the session to be gone but whoami returned a valid user id.
        // So throw this "error". This is the same as the case that the server is down.
        throw ...
    }
    // Now do something for the expected case: OpenSlides has to be shut down, the retry process will stop, redirect to login page ...
}); // no error callback so that the promise is rejected automaticly in this case (is this true, I am not sure)
```

Now I see a new problem: We send the whoami request and if the server is down we immediatelly try to connect to websocket. This fails. Then we wait 1000 ms for the next try. I think this is a strange flow ...